glean-core/src/database/mod.rs

```diff
@@ -843,20 +882,125 @@ mod test {
                 _ => panic!("Unexpected data found"),
             }
         };
-
         db.iter_store_from(Lifetime::Ping, test_storage, None, &mut snapshotter);
         assert_eq!(1, found_metrics, "We only expect 1 Lifetime.Ping metric.");
 
-        // Make sure data was **not** persisted with rkv.
-        let store: SingleStore = unwrap_or!(
-            db.rkv
-                .open_single(Lifetime::Ping.as_str(), StoreOptions::create()),
-            panic!()
-        );
-        let reader = unwrap_or!(db.rkv.read(), panic!());
+        // Make sure data **wasn't** persisted with rkv.
+        let store: SingleStore = db
+            .rkv
+            .open_single(Lifetime::Ping.as_str(), StoreOptions::create())
+            .unwrap();
+        let reader = db.rkv.read().unwrap();
         assert!(store
-            .get(&reader, &test_metric_id)
+            .get(&reader, format!("{}#{}", test_storage, test_metric_id))
             .unwrap_or(None)
             .is_none());
     }
+
+    #[test]
+    fn test_bulk_persistence_for_lifetime_ping() {
+        // Init the database in a temporary directory.
+        let dir = tempdir().unwrap();
+        let str_dir = dir.path().display().to_string();
+        let db = Database::new(&str_dir, true).unwrap();
+        let test_storage = "test-storage";
+
+        assert!(db.ping_lifetime_data.is_some());
+
+        // Attempt to record a known value.
+        let test_value1 = "test-value1";
+        let test_metric_id1 = "telemetry_test.test_name1";
+        db.record_per_lifetime(
+            Lifetime::Ping,
+            test_storage,
+            test_metric_id1,
+            &Metric::String(test_value1.to_string()),
+        )
+        .unwrap();
+
+        // Attempt to persist data.
+        db.persist_ping_lifetime_data().unwrap();
+
+        // Attempt to record another known value.
+        let test_value2 = "test-value2";
+        let test_metric_id2 = "telemetry_test.test_name2";
+        db.record_per_lifetime(
+            Lifetime::Ping,
+            test_storage,
+            test_metric_id2,
+            &Metric::String(test_value2.to_string()),
+        )
+        .unwrap();
+
+        {
```

Comment:
Can you comment on what we expect to find at this point? e.g. something along the lines of "We expect the first metric to be in rkv and the second to be recorded, but in memory, not in rkv, because.."

Reply:
Done :)

Also,
> Let's file a bug for adding that to the python bindings, no need to do it in this PR if it takes too much time.
Only now I saw that... I did it already.

Reply:
Np ;) I guess @mdboom has to re-review then :D